sm-core/src/main/java/com/salesmanager/core/business/modules/cms/category/digitalocean/DOSpaceCategoryImageFileManager.java

```diff
@@ -0,0 +1,246 @@
+  public static DOSpaceCategoryImageFileManager getInstance() {
+
+    if (fileManager == null) {
+      fileManager = new DOSpaceCategoryImageFileManager();
+    }
+
+    return fileManager;
+
+  }
+
+
+  @Override
+  public OutputContentFile getCategoryImage(CategoryImage categoryImage) throws ServiceException {
+    return null;
+  }
+
+  @Override
+  public OutputContentFile getCategoryImage(String categoryCode, String imageName) throws ServiceException {
+    return null;
+  }
+
+
+  @Override
+  public void removeCategoryImage(CategoryImage categoryImage) throws ServiceException {
+    try {
+      // get buckets
+      String bucketName = bucketName();
+
+      final AmazonS3 s3 = s3Client();
+
+      String nodePath = nodePath(categoryImage.getCategory().getCode(), String.valueOf(categoryImage.getId()));
+
+      s3.deleteObject(bucketName, nodePath + categoryImage.getCategoryImage());
+      LOGGER.info("Remove file");
+    } catch (final Exception e) {
+      LOGGER.error("Error while removing file", e);
+      throw new ServiceException(e);
+
+    }
+  }
+
+
+  @Override
+  public void addCategoryImage(CategoryImage categoryImage, ImageContentFile contentImage) throws ServiceException {
+    try {
+      // get buckets
+      String bucketName = bucketName();
+      final AmazonS3 s3 = s3Client();
+
+      String nodePath = nodePath(categoryImage.getCategory().getCode(), String.valueOf(categoryImage.getId()));
+
+      ObjectMetadata metadata = new ObjectMetadata();
+      metadata.setContentType(contentImage.getMimeType());
+
+      PutObjectRequest request = new PutObjectRequest(bucketName,
+              nodePath + categoryImage.getCategoryImage(), contentImage.getFile(), metadata);
+      request.setCannedAcl(CannedAccessControlList.PublicRead);
+
+
+      s3.putObject(request);
+
+
+      LOGGER.info("Category add image");
+
+    } catch (final Exception e) {
+      LOGGER.error("Error while adding image to category", e);
+      throw new ServiceException(e);
+
+    }
+  }
+
+  private Bucket getBucket(String bucket_name) {
+    final AmazonS3 s3 = s3Client();
+    Bucket named_bucket = null;
+    List<Bucket> buckets = s3.listBuckets();
+    for (Bucket b : buckets) {
+      if (b.getName().equals(bucket_name)) {
+        named_bucket = b;
+      }
+    }
+
+    if (named_bucket == null) {
+      named_bucket = createBucket(bucket_name);
+    }
+
+    return named_bucket;
+  }
+
+  private Bucket createBucket(String bucket_name) {
+    final AmazonS3 s3 = s3Client();
+    Bucket b = null;
+    if (s3.doesBucketExistV2(bucket_name)) {
+      System.out.format("Bucket %s already exists.\n", bucket_name);
+      b = getBucket(bucket_name);
+    } else {
+      try {
+        b = s3.createBucket(bucket_name);
+      } catch (AmazonS3Exception e) {
+        System.err.println(e.getErrorMessage());
+      }
+    }
+    return b;
+  }
+
+  /**
+   * Builds an amazon S3 client
+   * 
+   * @return
+   */
+  private AmazonS3 s3Client() {
```

Comment:
this client creation should be moved to a spring been.

Reply:
We can talk about it